tree.go

```diff
@@ -1384,9 +1384,9 @@ func (n *LeafNode) Delete(k []byte, _ NodeResolverFn) (bool, error) {
 
 		// Clear the corresponding commitment
 		if k[StemSize] < 128 {
-			n.c1 = nil
+			n.c1 = new(Point).SetIdentity()
 		} else {
-			n.c2 = nil
+			n.c2 = new(Point).SetIdentity()
```

Comment:
The new changes are aligned on how `NewLeafNode` works, right? [Look here](https://github.com/ethereum/go-verkle/blob/4bc28ef2ae62fc88d23df6b8d10afbeb87029cb8/tree.go#L289-L290).

Looks like in that `else` we aren't doing `else if count > 0 {`, which means `cfg.CommitToPoly` is always called which never return `nil`.

So maybe this means the current change is a more reasonable action that what we had before this PR?